File: Algorithms/C++/1313-Decompress_Run-Length_Encoded_List.cpp
// Runtime: 52 ms, faster than 11.52% of C++ online submissions for Decompress Run-Length Encoded List.
// Memory Usage: 10.7 MB, less than 100.00% of C++ online submissions for Decompress Run-Length Encoded List.
class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        vector<int> res;
        for (int i = 0; i < nums.size(); i += 2)
            for (int j = 0; j < nums[i]; ++j)
                res.push_back(nums[i + 1]);
        return res;
    }
};

File: Algorithms/Python3.x/1313-Decompress_Run-Length_Encoded_List.py
# Runtime: 84 ms, faster than 5.74% of Python3 online submissions for Decompress Run-Length Encoded List.
# Memory Usage: 13 MB, less than 100.00% of Python3 online submissions for Decompress Run-Length Encoded List.
class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        res = []
        for idx in range(0, len(nums), 2):
            for _ in range(nums[idx]):
                res.append(nums[idx + 1])
        return res
